app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -824,8 +824,18 @@ def prepare_vm_disk_attachment(disk_spec, storage_spec)
     #
     def remove_vm_disks(vm_service, disk_specs)
       attachments_service = vm_service.disk_attachments_service
+
+      disk_attachment_by_disk_name_hash = attachments_service.list.collect do |attachment_service_obj|
+        attachment_service = attachments_service.attachment_service(attachment_service_obj.id)
+        [vm_service.connection.follow_link(attachment_service_obj.disk).name, attachment_service]
+      end.to_h
+
       disk_specs.each do |disk_spec|
-        attachment_service = attachments_service.attachment_service(disk_spec['disk_name'])
+        disk_spec = disk_spec.with_indifferent_access
```

Comment:
@borod108 just curious, why do you need with_indifferent_access here?  Is the `'disk_name'` key not always a string ?

Reply:
When the request comes from api it is, at least sometimes, a symbol (this was one of the things that went wrong here)

Reply:
Interesting ok thanks